fix: tag regex no longer includes purely numerical 'tags'

wutscher opened the pull request:
Changed the tag regex to not include tags which only consist of numbers. This is to be consistent with the way obsidian does it.

Example:
```js
#test  //tag
#test1 //tag
#1test //tag
#12345 //not a tag
```

wutscher committed:
fix: tag regex no longer includes purely numerical 'tags'

wutscher committed:
fix: formatting

jackyzha0 reviewed (commented):
- Thanks! I think the regex is already really messy here so would prefer to put this additional logic in the `findAndReplace(tree, tagRegex, (_value: string, tag: string) => {` block around L407. If you return `false` in that function if the match only contains numbers, it should just skip the replace.

more on the obsidian spec: https://help.obsidian.md/Editing+and+formatting/Tags#Tag+format

wutscher committed:
fix: use guard in findAndReplace() instead of expanding the regex

wutscher commented:
I reverted the regex back to it's original form and moved the logic to the `findAndReplace()` function.

Screenshot of the new behaveure:
![image](https://github.com/jackyzha0/quartz/assets/25224756/7b5e3be4-c6dd-4b0e-b3c0-b4564c8ea850)
